include/graaflib/algorithm/shortest_path/floyd_warshall.tpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <graaflib/algorithm/shortest_path/floyd_warshall.h>
+
+#include <limits>
+#include <vector>
+
+namespace graaf::algorithm {
+
+namespace detail {
+
+// Mark vertices in negative cycle as NO_PATH
+template <typename WEIGHT_T>
+void do_fw_detect_negative_cycle(std::vector<std::vector<WEIGHT_T>>& distance,
+                                 WEIGHT_T ZERO, WEIGHT_T INF, WEIGHT_T NO_PATH,
+                                 size_t n) {
+  for (std::size_t start_vertex = 0; start_vertex < n; ++start_vertex) {
+    for (std::size_t end_vertex = 0; end_vertex < n; ++end_vertex) {
+      for (std::size_t through_vertex = 0; through_vertex < n;
+           ++through_vertex) {
+        if (distance[start_vertex][through_vertex] < INF &&
+            distance[through_vertex][through_vertex] < ZERO &&
+            distance[through_vertex][end_vertex] < INF) {
+          distance[start_vertex][end_vertex] = NO_PATH;
+        }
+      }
+    }
+  }
+}
+
+};  // namespace detail
+
+template <typename V, typename E, graph_type T, typename WEIGHT_T>
+std::vector<std::vector<WEIGHT_T>> floyd_warshall_shortest_paths(
+    const graph<V, E, T>& graph) {
+  WEIGHT_T ZERO{};
+  std::size_t n = graph.vertex_count();
+  auto INF = std::numeric_limits<WEIGHT_T>::max();
+  auto NO_PATH = std::numeric_limits<WEIGHT_T>::min();
```

Comment:
Regarding using `std::numeric_limits<WEIGHT_T>::min()` for vertex pairs between which we cannot find a path (i.e. vertices in negative weight cycles), consider the case where the `WEIGHT_T` is an `unsigned` type. 

In that case, the caller cannot distinguish between a `NO_PATH` and a true distance of `0`. However, I do not see an elegant way of handling this. What do you think about removing the negative cycle detection, as it is also documented on the interface that the algorithm does not work in this case.

Reply:
Oh, I didn't see this one. My bad. I'll fix it right away. Agreed, we could move it to a different PR as Floyd-Warshall negative cycle detection so results will be less confusing. And in this case, NO_PATH will no longer be needed.

I look forward to your next comments about PR. :)